src/lib/viewers/doc/BoxExcel/sheet.js

```diff
@@ -0,0 +1,640 @@
+                    width: totalWidth
+                };
+            });
+        }
+
+        return {
+            rowCount,
+            columnCount,
+            rowHeight,
+            columnWidth,
+            rowHeights,
+            columnWidths,
+            merges
+        };
+    };
+
+    /**
+     * Get data object for a certain position
+     *
+     * @param  {number} rowIndex    row index, 1 basis
+     * @param  {number} columnIndex col index, 1 basis
+     * @return {Object}             data object for the cell
+     */
+    _getCell = (rowIndex, columnIndex) => {
+        /* global XLSX */
+        const pos = XLSX.utils.encode_cell({
+            r: rowIndex - 1,
+            c: columnIndex - 1
+        });
+        const { sheet } = this.state;
+        return sheet[pos];
+    };
+
+    /**
+     * Helper function of _getBorder
+     *
+     * @param  {Object} style style property in a cell object
+     * @return {string}       hex color string
+     */
+    _getBorderColor = (style) => {
+        return style && style.color ? _parseColor(style.color) : c.gridGrey;
+    };
+
+    /**
+     * Helper function of _getBorder
+     *
+     * @param  {style} style style property in a cell object
+     * @return {string}      border width string, default: '1px'
+     */
+    _getBorderWidth = (style) => {
+        return style && style.style && borderWidthMap[style.style] ? borderWidthMap[style.style] : '1px';
+    };
+
+    /**
+     * Get border style, including borderColor and borderWidth
+     *
+     * @param  {Object} style style property in a cell object
+     * @return {Object}       React style object
+     */
+    _getBorder = (style) => {
+        const borderColor = `${this._getBorderColor(style.top)} ${this._getBorderColor(
+            style.right
+        )} ${this._getBorderColor(style.bottom)} ${this._getBorderColor(style.left)}`;
+        const borderWidth = `${style.top ? this._getBorderWidth(style.top) : '0px'} ${
+            style.right ? this._getBorderWidth(style.right) : '1px'
+        } ${style.bottom ? this._getBorderWidth(style.bottom) : '1px'} ${
+            style.left ? this._getBorderWidth(style.left) : '0px'
+        }`;
+        return { borderColor, borderWidth };
+    };
+
+    /**
+     * Get cell content
+     *
+     * @param  {number} rowIndex    row index, 1 basis, 0 is header
+     * @param  {number} columnIndex col index, 1 basis, 0 is header
+     * @return {string}             cell content
+     */
+    _getCellContent = (rowIndex, columnIndex) => {
+        if (rowIndex === 0 && columnIndex === 0) {
+            return '';
+        } else if (rowIndex === 0) {
+            /* global XLSX */
+            return XLSX.utils.encode_col(columnIndex - 1);
+        } else if (columnIndex === 0) {
+            return rowIndex;
+        }
+        const cell = this._getCell(rowIndex, columnIndex);
+        if (!cell) return '';
+        if (cell.t === 'd') {
```

Comment:
Pull these out into variables so that it's easier to understand what t, d, w are